src/fragments/lib/graphqlapi/ios/mutate-data.mdx

```diff
@@ -48,21 +48,23 @@ func updateTodo() -> AnyCancellable {
     // Retrieve your Todo using Amplify.API.query
     var todo = Todo(name: "my first todo", description: "todo description")
     todo.description = "updated description"
-    let sink = Amplify.API.mutate(request: .update(todo))
-        .resultPublisher
-        .sink {
-            if case let .failure(error) = $0 {
-                print("Got failed event with error \(error)")
-            }
+    let todoUpdated = todo
```

Comment:
is there a reason to create a new variable here? how about this where we clarify the "retrievedTodo" and then a `var updatedTodo` that we modify?
```
let retrievedTodo = Todo(name: "my first todo", description: "todo description")
    todo.description = "updated description")
var updatedTodo = retrievedTodo
updatedTodo.description = "updatedDescription"
```

Reply:
The `async/await` block doesn't allow `toDo` to be a `var` and gives the error `Reference to captured var in a concurrently executing code`